libvirt/resource_libvirt_domain.go

```diff
@@ -106,6 +118,34 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		domainDef.Metadata.TerraformLibvirt.Xml = metadata.(string)
 	}
 
+	if firmware, ok := d.GetOk("firmware"); ok {
+		firmwareFile, err := homedir.Expand(firmware.(string))
```

Comment:
I don't understand why we need `homedir` here. Is it because a user could set `firmware = "~/foo"`? If that's the case I would just remove this dependency and force the user to write the full path (like with all the configuration files used by other programs)